### PR TITLE
Benchmarks broadcast update

### PR DESCRIPTION
**Pull request by csarofeen**
Adding broadcast benchmarks.
Expanding range of sizes in BatchNorm benchmark.
Increased multipliers for both of these benchmarks.
Update scale bias relu to new benchmark mechanisms.

### CODE REVIEW

File: benchmarks/cpp/nvfuser/broadcast.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                        .dtype(dtype)
+                        .contiguity({true, true})
+                        .build();
+  fusion->addInput(tv0);
+
+  std::vector<bool> bcast_pattern(2, false);
+  bcast_pattern[bcast_axis] = true;
+
+  TensorView* tv1 =
+      TensorViewBuilder().ndims(1).dtype(dtype).contiguity({true}).build();
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/broadcast.cpp]: Replace with `makeContigTensor`

File: benchmarks/cpp/nvfuser/broadcast.cpp [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+using namespace torch::jit::fuser::cuda;
+
+// Return broadcast tensor view and output of broadcast
+static void setupBroadcast(Fusion* fusion, DataType dtype, int bcast_axis) {
+  FusionGuard fg(fusion);
+
+  bool is_fp16 = dtype == DataType::Half;
+
+  TensorView* tv0 = TensorViewBuilder()
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/broadcast.cpp]: Replace with `makeContigTensor`

File: benchmarks/cpp/nvfuser/scale_bias_relu.cpp [hunk 3]
```diff
@@ -12,61 +12,102 @@
...
-  for (size_t axis = 0; axis < kNumberOfDims - 1; ++axis) {
-    broadcast_mask[axis] = true;
-  }
+  std::vector<int64_t> bcast_shape(kNumberOfDims, 1);
+  bcast_shape[bcast_shape.size() - 1] = -1;
+
+  std::vector<bool> bcast_contig(kNumberOfDims, false);
+  bcast_contig[bcast_contig.size() - 1] = true;
+
+  auto x = TensorViewBuilder()
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/scale_bias_relu.cpp]: Replace with `makeContigTensor`

File: benchmarks/cpp/nvfuser/scale_bias_relu.cpp [hunk 4]
```diff
@@ -12,61 +12,102 @@
...
-  fusion->addInput(bias_half);
-  fusion->addInput(mean_half);
-  fusion->addInput(var_half);
+  const size_t kNumberOfDims = 4;
 
-  std::vector<bool> broadcast_mask(kNumberOfDims, false);
-  for (size_t axis = 0; axis < kNumberOfDims - 1; ++axis) {
-    broadcast_mask[axis] = true;
-  }
+  auto x = TensorViewBuilder()
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/scale_bias_relu.cpp]: Replace with `makeContigTensor`

**Review by rdspring1 — APPROVED** [review]
LGTM.

We could incorporate `makeContigTensor` so we are consistent with test_gpu.cpp
```
// Make a tensor that is known to be fully contiguous of dimensionality=ndims,
// but unknown sizes
TensorView* makeContigTensor(size_t ndims, DataType dtype = DataType::Float) {
  return TensorViewBuilder()
      .ndims(ndims)
      .dtype(dtype)
      .contiguity(std::vector<bool>(ndims, true))
      .build();
}
```

File: benchmarks/cpp/nvfuser/gelu_backward.cpp [hunk 5]
```diff
@@ -23,19 +23,19 @@ static void setupFusion(Fusion* fusion) {
   const float k_010 = 0.1070322243;
 
   // gradient tensor
-  auto t0 = TensorViewBuilder().ndims(3).dtype(DataType::Half).build();
+  auto t0 = makeContigTensor(3, DataType::Half);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/gelu_backward.cpp:26]: Since the contiguity wasn't originally specified, should this be `makeSymbolicTensor`?

```
// Make a tensor that is known to be non-contiguous of dimensionality=ndims,
// but unknown sizes
TensorView* makeSymbolicTensor(size_t ndims, DataType dtype = DataType::Float) {
  return TensorViewBuilder().ndims(ndims).dtype(dtype).build();
}
```

File: benchmarks/cpp/nvfuser/instance_norm.cpp [hunk 6]
```diff
@@ -18,13 +18,11 @@ static void setupInstanceNorm(Fusion* fusion, DataType dtype) {
...
   FusionGuard fg(fusion);
 
-  auto input = TensorViewBuilder().ndims(4).dtype(dtype).build();
-  auto weight = TensorViewBuilder().ndims(1).dtype(dtype).build();
-  auto bias = TensorViewBuilder().ndims(1).dtype(dtype).build();
-  auto running_mean =
-      TensorViewBuilder().ndims(1).dtype(DataType::Float).build();
-  auto running_var =
-      TensorViewBuilder().ndims(1).dtype(DataType::Float).build();
+  auto input = makeContigTensor(4, dtype);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/instance_norm.cpp:21]: Maybe `makeSymbolicTensor`

File: benchmarks/cpp/nvfuser/layer_norm.cpp [hunk 7]
```diff
@@ -27,9 +27,10 @@ static void setupLayerNorm(Fusion* fusion, DataType dtype) {
   Double* eps_ptr = new Double(kEps);
 
   // setup fusion
-  auto input = TensorViewBuilder().ndims(2).dtype(dtype).build();
-  auto weight = TensorViewBuilder().ndims(1).dtype(dtype).build();
-  auto bias = TensorViewBuilder().ndims(1).dtype(dtype).build();
+  auto input = makeContigTensor(2, dtype);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/layer_norm.cpp:30]: Maybe `makeSymbolicTensor`

File: benchmarks/cpp/nvfuser/lstm_cell.cpp [hunk 8]
```diff
@@ -20,15 +20,11 @@ static void setupFusion(Fusion* fusion) {
 
   TensorView* tvs[16];
   for (size_t i = 0; i < 16; i++) {
-    tvs[i] = TensorViewBuilder().ndims(2).dtype(DataType::Float).build();
+    tvs[i] = makeContigTensor(2, DataType::Float);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/lstm_cell.cpp:23]: Maybe `makeSymbolicTensor`

File: benchmarks/cpp/nvfuser/softmax.cpp [hunk 9]
```diff
@@ -25,7 +25,7 @@ static void setupSoftmax(
 
   FusionGuard fg(fusion);
   // setup fusion
-  auto input = TensorViewBuilder().ndims(2).dtype(dtype).build();
+  auto input = makeContigTensor(2, dtype);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/softmax.cpp:28]: Maybe `makeSymbolicTensor`

File: benchmarks/cpp/nvfuser/scale_bias_relu.cpp [hunk 10]
```diff
@@ -12,62 +12,81 @@
...
-  fusion->addInput(bias_half);
-  fusion->addInput(mean_half);
-  fusion->addInput(var_half);
-
-  std::vector<bool> broadcast_mask(kNumberOfDims, false);
-  for (size_t axis = 0; axis < kNumberOfDims - 1; ++axis) {
-    broadcast_mask[axis] = true;
+  const size_t kNumberOfDims = 4;
+
+  auto x = makeContigTensor(kNumberOfDims, dtype);
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/scale_bias_relu.cpp:67]: Maybe `makeSymbolicTensor`
